tests/invalid_models_tests/test_models.py

```diff
@@ -761,3 +761,25 @@ class ParkingLot(Place):
             'as an implicit link is deprecated.'
         )
         self.assertEqual(ParkingLot._meta.pk.name, 'parent')
+
+    def test_m2m_table_name_clash(self):
+        class Foo(models.Model):
+            bar = models.ManyToManyField('Bar', db_table='myapp_bar')
+
+            class Meta:
+                db_table = 'myapp_foo'
+
+        class Bar(models.Model):
+            class Meta:
+                db_table = 'myapp_bar'
+
+        errors = Foo.check()
+        expected = [
+            Error(
+                "Field is using a table that has already been registered by "
+                "<class 'invalid_models_tests.test_models.OtherModelTests.test_m2m_table_name_clash.<locals>.Bar'>.",
```

Comment:
This assertion will fail on Python 2 as the `repr`esentation of locally defined classes is not the same.